src/txadmission.cpp

```diff
@@ -253,31 +243,76 @@ void CommitTxToMempool()
             SyncWithWallets(data.entry.GetSharedTx(), nullptr, -1);
 #endif
             vWhatChanged.push_back(data.hash);
-
             // Update txn per second only when a txn is valid and accepted to the mempool
             mempool.UpdateTransactionsPerSecond();
         }
         txCommitQ.clear();
     }
 
+    std::map<uint256, CTxInputData> mapWasDeferred;
     {
         LOCK(csTxInQ);
         // Clear the filter of incoming conflicts, and put all queued tx on the deferred queue since they've been
         // deferred
+        LOG(MEMPOOL, "txadmission incoming filter reset.  Current txInQ size: %d\n", txInQ.size());
         incomingConflicts.reset();
         while (!txInQ.empty())
         {
             txDeferQ.push(txInQ.front());
             txInQ.pop();
         }
-        LOG(MEMPOOL, "Reset incoming filter\n");
+
+        // Move the previously deferred txns into active processing.
+
+        // We MUST push the first item in the defer queue in without checking it against incoming conflicts.
+        // This is fine because the first insert into an empty incomingConflicts must succeed.
+        // A transaction's inputs could cause a false positive match against each other.  By pushing the first
+        // deferred tx without checking, we can still use the efficient fastfilter checkAndSet function for most queue
+        // filter checking but mop up the extremely rare tx whose inputs have false positive matches here.
+        if (!txDeferQ.empty())
+        {
+            const CTxInputData &first = txDeferQ.front();
+
+            for (auto inp : first.tx->vin)
+            {
+                uint256 hash = IncomingConflictHash(inp.prevout);
+                incomingConflicts.insert(hash);
+            }
```

Comment:
I don't call EnqueueTxForAdmission() for the first tx.  I insert all its input into the incomingConflcts FF and push it onto the txInQ directly.  So we won't ever end up doing it again...

Reply:
I know, but then aren't you also calling incomingConflicts.insert() for every following txn rather than letting that happen later using checkAndSet() ?

Reply:
no I only call insert for the first deferred txn.  To guess at why you are thinking I am doing it for all: the for loop is iterating through all of the inputs in that one txn.

Reply:
ah yes, i see that you're draining the defer queue right after that...